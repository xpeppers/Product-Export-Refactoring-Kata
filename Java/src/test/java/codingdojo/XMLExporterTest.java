package codingdojo;

import org.approvaltests.Approvals;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Collection;

import static codingdojo.SampleModelObjects.*;
import static java.util.Collections.*;

public class XMLExporterTest {

    @Test
    public void exportFull_OneOrder() {
        Collection<Order> orderList = new ArrayList<>();
        orderList.add(SampleModelObjects.RecentOrder);

        String result = XMLExporter.exportFull(orderList);

        Approvals.verifyXml(result);
    }

    @Test
    public void exportFull_ZeroOrders() {
        String result = XMLExporter.exportFull(emptyList());

        Approvals.verifyXml(result);
    }

    @Test
    public void exportFull_OrderWithNoEventProduct() {
        Product productWithNoEvent = CherryBloom;
        Order order = new Order("1234", Util.fromIsoDate("2018-09-01T00:00Z"),
                FlagshipStore, new Product[]{productWithNoEvent});
        String result = XMLExporter.exportFull(singletonList(order));

        Approvals.verifyXml(result);
    }
}
