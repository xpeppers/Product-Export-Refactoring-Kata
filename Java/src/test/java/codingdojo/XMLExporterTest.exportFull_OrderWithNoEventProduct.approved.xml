<?xml version="1.0" encoding="UTF-8"?><orders>
  <order id="1234">
    <product id="LIPSTICK01" weight="30">
      <price currency="USD">14.99</price>Cherry Bloom</product>
  </order>
</orders>
